src/foam/util/SafetyUtil.java

```diff
@@ -60,6 +60,12 @@ public static int compare(Object o1, Object o2) {
     return ((Comparable) o1).compareTo(o2);
   }
 
+  public static int compare(Object[] o1, Object[] o2) {
+    if ( o2 == null ) return  1;
+    if ( o1 == null ) return -1;
+    return compare(o1.length, o2.length);
```

Comment:
Shouldn't we also compare their contents?

Reply:
@kgrgreer comparing contents only works if the order is guaranteed.